lib/ui/text.dart

```diff
@@ -431,6 +431,85 @@ enum TextDecorationStyle {
   wavy
 }
 
+/// Defines how the paragraph will handle the ascent of the first line and
+/// descent of the last line. These lines are referred to as "boundary" lines.
+class BoundaryLineHeightBehavior {
```

Comment:
by calling this _boundary_ line height behaviour you're boxing yourself in in case you ever want to e.g. have different behaviour for odd lines vs even lines. Can we call it something more general? TextHeightBehavior or something?

Reply:
Maybe just `LineHeightBehavior`? `TextHeightBehavior` seems a bit redundant. Or calling it just `HeightBehavior` allows it to be flexible enough to add all sorts of modifications.